.github/workflows/release.yml

```diff
@@ -157,22 +160,26 @@ jobs:
         echo "Installed Linkerd CLI version: $TAG"
         echo "::set-env name=CMD::$CMD"
     - name: Set KUBECONFIG environment variables
+      if: startsWith(github.ref, 'refs/tags/stable')
       run: |
         mkdir -p $HOME/.kube
         echo "${{ secrets.ARM64_KUBECONFIG }}" > $HOME/.kube/config
         echo ::set-env name=KUBECONFIG::$HOME/.kube/config
         kubectl cluster-info
     - name: Run integration tests
+      if: startsWith(github.ref, 'refs/tags/stable')
       env:
         RUN_ARM_TEST: 1
       run: bin/tests --skip-kind-create "$CMD"
     - name: CNI tests
+      if: startsWith(github.ref, 'refs/tags/stable')
       run: |
         export TAG="$($CMD version --client --short)"
         go test -cover -race -v -mod=readonly ./cni-plugin/test -integration-tests
     - name: Test cleanup
       if: ${{ always() }}
-      run: bin/test-cleanup
+      # will fail if other steps didn't run, so ignore error
+      run: bin/test-cleanup || true
```

Comment:
I think this step can also be marked with [continue-on-error](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error)